api/agent/lb_agent.go

```diff
@@ -227,18 +227,19 @@ func (a *lbAgent) placeDetachCall(ctx context.Context, call *call) error {
 }
 
 func (a *lbAgent) placeCall(ctx context.Context, call *call) error {
-	err := a.placer.PlaceCall(a.rp, ctx, call)
+	err := a.placer.PlaceCall(a.rp, ctx, call, a.placer.Config().PlacerTimeout)
```

Comment:
we don't need to add the parameter to the placer interface - we can pass a context that has placerTimeout+callTimeout from right here to encode that info when applicable (when detached)

Reply:
@rdallman I'm ok with reducing arguments here, but if you actually meant setting a deadline for that ctx, then it's a bit tricky to do that using that ctx. Here ctx is a detached client that's kept alive by LB using some magic timeout numbers (see above) to set an upper bound of total wait time. Placer timeout however is a subset (special) timer that's only active when searching for a runner. This timer  gets ignored if a runner is found.

Reply:
just a few lines later https://github.com/fnproject/fn/pull/1267/files#diff-a4aff9524fa41710d2ab6943f2bbc713R243 it ends up doing this

Reply:
Math in timeout calc in there is wrong.

Reply:
as pointed out by @skinowski we can't pass just via the context as we need to differentiate between the detach placement timeout and the request timeout. I don't feel terrible to add the placertimeout to the PlaceCall interface, this change is all about set different timeouts so the single placeTimeout is not enough anymore. I can add a method to the placer to set the placerTimeout and use just that timeout in the PlacerTracker but to me that makes the code less readable.